motor/discovery/azure/mql_assets.go

```diff
@@ -0,0 +1,226 @@
+package azure
+
+import (
+	"context"
+	"errors"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/rs/zerolog/log"
+	"go.mondoo.com/cnquery"
+	"go.mondoo.com/cnquery/llx"
+	"go.mondoo.com/cnquery/motor"
+	"go.mondoo.com/cnquery/motor/asset"
+	"go.mondoo.com/cnquery/motor/discovery/common"
+	"go.mondoo.com/cnquery/motor/platform"
+	"go.mondoo.com/cnquery/motor/providers"
+	"go.mondoo.com/cnquery/motor/vault"
+	"go.mondoo.com/cnquery/mql"
+	"go.mondoo.com/cnquery/resources"
+
+	azure "go.mondoo.com/cnquery/motor/providers/microsoft"
+	"go.mondoo.com/cnquery/motor/providers/resolver"
+	resource_pack "go.mondoo.com/cnquery/resources/packs/azure"
+)
+
+type MqlDiscovery struct {
+	rt *resources.Runtime
+}
+
+func NewMQLAssetsDiscovery(provider *azure.Provider) (*MqlDiscovery, error) {
+	m, err := motor.New(provider)
+	if err != nil {
+		return nil, err
+	}
+	rt := resources.NewRuntime(resource_pack.Registry, m)
+	return &MqlDiscovery{rt: rt}, nil
+}
+
+func (md *MqlDiscovery) Close() {
+	if md.rt != nil && md.rt.Motor != nil {
+		md.rt.Motor.Close()
+	}
+}
+
+func GetList[T any](md *MqlDiscovery, query string) ([]T, error) {
+	mqlExecutor := mql.New(md.rt, cnquery.DefaultFeatures)
+	value, err := mqlExecutor.Exec(query, map[string]*llx.Primitive{})
+	if err != nil {
+		return nil, err
+	}
+	if value.Error != nil {
+		return nil, value.Error
+	}
+	var out []T
+	if err := mapstructure.Decode(value.Value, &out); err != nil {
+		return nil, err
+	}
+	return out, nil
+}
+
+func GatherAssets(ctx context.Context, tc *providers.Config, credsResolver vault.Resolver, sfn common.QuerySecretFn) ([]*asset.Asset, error) {
+	assets := []*asset.Asset{}
+	pCfg := tc.Clone()
+	motor, err := resolver.NewMotorConnection(ctx, pCfg, credsResolver)
+	if err != nil {
+		return nil, err
+	}
+	defer motor.Close()
+
+	provider, ok := motor.Provider.(*azure.Provider)
+	if !ok {
+		return nil, errors.New("could not create azure provider")
+	}
+	m, err := NewMQLAssetsDiscovery(provider)
+	if err != nil {
+		return nil, err
+	}
+	// TODO: make this work by having the ip addresses available via mql
+	// if tc.IncludesOneOfDiscoveryTarget(common.DiscoveryAll, DiscoveryInstances) {
+	// 	instances, err := computeInstances(m, provider.SubscriptionID(), pCfg)
+	// 	if err != nil {
+	// 		return nil, err
+	// 	}
+	// 	assets = append(assets, instances...)
+	// }
+	if tc.IncludesOneOfDiscoveryTarget(common.DiscoveryAll, DiscoverySqlServers) {
```

Comment:
For all these discovery target checks:
Somewhere below you have the option `common.DiscoveryAuto`. Should any of these assets be part of the auto discovery.

Reply:
Not right now, only subs and instances are part of the auto discovery. We will enable this at a later point